### DISCUSSION TITLE
Dart API join with NNBD

### DISCUSSION THREAD

**Discussion by davidmartos96** [simolus3/drift#1076] [Q&A] [answered]
Hello @simolus3 !
I'm trying to port some of my code to null safety and I encountered with the following case:

```dart
leftOuterJoin(
     myTable,
     myTable.nonNullCol.equalsExp(otherTable.nonNullCol) &
            myTable.otherNonNullCol.equals(0),
     useColumns: false,
),
``` 

The migration tool recommended to cast the input Expression for the join function as `Expression<bool>`, because the output of the "AND" is a `Expression<bool?>`. I was wondering if this is a safe cast that can be done or if there is a better approach to tackle these situations. I know there can be issues when casting Lists with nullable items as Lists with non nullable items, that's why I'm asking.
Thanks in advance!!

**Comment by simolus3** [comment, marked as answer]
I think I just missed that case in `leftOuterJoin` and `innerJoin` - an `Expression<bool?>` is the right type for now.

This is fixed on `develop`, you can use this as a workaround until then:

```dart
leftOuterJoin(
     myTable,
     (myTable.nonNullCol.equalsExp(otherTable.nonNullCol) &
            myTable.otherNonNullCol.equals(0)).dartCast(),
     useColumns: false,
),
```

**Reply by davidmartos96** [reply]
Thanks @simolus3, that works!

**Comment by rubenferreira97** [comment]
@simolus3 I am having some problems with NNBD on the `onConflict` clause:
Original:
` onConflict: DoUpdate((old) => StatisticsTableCompanion.custom(cliques: old.cliques + Constant(statistic.cliques))));`

Does this work as a temporary solution? This is not giving me any compile errors but I still did not test it.
` onConflict: DoUpdate((old) => StatisticsTableCompanion.custom(cliques: (old.cliques + Constant(statistic.cliques)).dartCast())));`
